src/hotspot/share/opto/cfgnode.cpp

```diff
@@ -1117,9 +1117,35 @@ const Type* PhiNode::Value(PhaseGVN* phase) const {
           if (bt != BoolTest::ne) {
             if (stride_t->hi_as_long() < 0) {          // Down-counter loop
               swap(lo, hi);
-              return TypeInteger::make(MIN2(lo->lo_as_long(), hi->lo_as_long()), hi->hi_as_long(), 3, l->bt())->filter_speculative(_type);
+              jlong first = lo->lo_as_long();
+              if (first < max_signed_integer(l->bt())) {
+                first += 1; // lo is after decrement
+                // When bounds are constant and ABS(stride) greater than 1, exact bounds for the phi can be computed
+                if (lo->is_con() && hi->is_con() && hi->lo_as_long() > lo->hi_as_long() && stride_t->lo_as_long() != -1) {
```

Comment:
`stride` is constant. May be used the value instead of calling `stride_t->lo_as_long()` 3 times in this code.